amethyst_network/src/net_event.rs

```diff
@@ -3,38 +3,19 @@
 //! NetOwnedEvent are passed through the ECS, and contains the event's source (remote connection, usually).
 
 use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
 use uuid::Uuid;
 
 /// Network events which you can send or and receive from an endpoint.
+// TODO, Connect, connection refused, disconnect, disconnected
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub enum NetEvent<T> {
-    /// Ask to connect to the server.
-    Connect {
-        /// The client uuid.
-        client_uuid: Uuid,
-    },
-    /// Reply to the client that the connection has been accepted.
-    Connected {
-        /// The server uuid.
-        server_uuid: Uuid,
-    },
-    /// Reply to the client that the connection has been refused.
-    ConnectionRefused {
-        /// The reason of the refusal.
-        reason: String,
-    },
-    /// Tell the server that the client is disconnecting.
-    Disconnect {
-        /// The reason of the disconnection.
-        reason: String,
-    },
-    /// Notify the clients(including the one being disconnected) that a client has been disconnected from the server.
-    Disconnected {
-        /// The reason of the disconnection.
-        reason: String,
-    },
+    Connected(SocketAddr),
+    Disconnected(SocketAddr),
     /// Send a packet to all connected clients
     Packet(NetPacket<T>),
+    #[doc(hidden)]
+    __Nonexhaustive,
```

Comment:
Why?

Reply:
Because more events will come, this will make us a bit more future proof.